CSharpMath.Editor/Extensions/MathList.cs

```diff
@@ -81,12 +83,66 @@ partial class Extensions {
     }
 
     public static void RemoveAt(this MathList self, ref MathListIndex index) {
-      index ??= MathListIndex.Level0Index(0);
+      void RemoveAtInnerList<TAtom>(ref MathListIndex index, TAtom atom, int innerListIndex) where TAtom : MathAtom, IMathListContainer {
```

Comment:
Can we document this a bit. For example it's not clear from the name of the function and inputs:
- Is `atom` an Atom that resides at `index`?
- Is `atom` the thing to be removed, or is instead a subatom of `atom` residing at `innerListIndex` within `atom`?

Reply:
`atom` is a `MathAtom` directly contained in `self` that contains/is the atom to remove.

Reply:
OK that takes us closer. What would be the right type for AtomIndex to have? `Nullable<int>`?

Reply:
An AtomIndex of `-1` indicates deletion at the beginning of line (not pointing to any atom). It is not a valid value for other methods accepting `MathListIndex`.